prog/weaveutil/plugin_network.go

```diff
@@ -71,6 +73,34 @@ while Weave is not running`, networkName, containers)
 	return nil
 }
 
+// Exits with 0 if the given plugin (v2) is enabled.
+func isDockerPluginEnabled(args []string) error {
+	if len(args) != 1 {
+		cmdUsage("is-docker-plugin-enabled", "<plugin-name>")
+	}
+
+	pluginName := args[0]
+
+	// This is messed up: we are using docker/docker/client instead of
+	// fsouza/go-dockerclient because the latter does not support plugins.
+	c, err := client.NewEnvClient()
+	if err != nil {
+		return fmt.Errorf("unable to connect to docker: %s", err)
+	}
+
+	ctx := context.Background()
+	p, _, err := c.PluginInspectWithRaw(ctx, pluginName)
```

Comment:
the cmd which calls the noted function is invoked only from `is_plugin_v2_enabled` in the weave script. so, in the case of the older Docker version, the noted function should fail, and no plugin-v2 will be detected. this is fine as the older version does not support v2 plugins anyway.